### DISCUSSION TITLE
--tests switch not functioning correctly

### DISCUSSION THREAD

**Discussion by huornlmj** [golangci/golangci-lint#2481] [Q&A] [answered]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).
- [X] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

### Description of the problem

Using golangci-lint both with and without the --tests switch scans *test.go files regardless.  The help file says...

> --tests                          Analyze tests (*_test.go) (default true)

Suggests that if --tests is present then *_test.go files are scanned.  But the default is true, so what is supposed to happen if it's omitted if the default is true?

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 1.41.1 built from a2074809 on 2021-06-19T16:01:50Z
```

</details>

### Configuration file

None

### Go environment

<details>

```console
$ go version && go env
go version go1.16.3 linux/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN="/home/REDACTED/go/bin"
GOCACHE="/home/REDACTED/.cache/go-build"
GOENV="/home/REDACTED/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/REDACTED/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/REDACTED/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/lib/go-1.16"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/lib/go-1.16/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.16.3"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/REDACTED/source/REDACTED/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build3891235750=/tmp/go-build -gno-record-gcc-switches"
```

</details>

### Verbose output of running

None

### Code example or link to a public repository

None

**Comment by ldez** [comment, marked as answer]
Hello,

First, I recommend not using flags.

The default is `true` so if you omit the flag or if you are using `--test` or if you are using `--test true`, the test files are checked.

If you want to disable that you have to use `--test false`

**Reply by huornlmj** [reply]
Ok that explains it, so perhaps it might be prudent to add the true|false options in the --help test for the --tests switch?
